cmd/subnetcmd/publish.go

```diff
@@ -0,0 +1,380 @@
+
+func getSubnetInfo(sc *models.Sidecar) (*types.Subnet, error) {
+	homepage, err := app.Prompt.CaptureEmpty("What is the homepage of the Subnet project?", nil)
+	if err != nil {
+		return nil, err
+	}
+
+	desc, err := app.Prompt.CaptureEmpty("Please provide a free-text description of the Subnet", nil)
+	if err != nil {
+		return nil, err
+	}
+
+	maintrs, canceled, err := app.Prompt.CaptureListDecision(
+		app.Prompt,
+		"Who are the maintainers of the Subnet?",
+		app.Prompt.CaptureEmail,
+		"Provide a maintainer",
+		"Maintainer",
+		"",
+		nil,
+	)
+	if err != nil {
+		return nil, err
+	}
+	if canceled {
+		ux.Logger.PrintToUser("Publishing aborted")
+		return nil, errors.New("canceled by user")
+	}
+
+	strMaintrs := make([]string, len(maintrs))
+	for i, m := range maintrs {
+		strMaintrs[i] = m.(string)
+	}
+
+	// TODO: In this version, we are publishing 1 subnet and exactly 1 VM.
+	// 1. Will the VM list has to be editale in the future?
+	// 2. If yes, need to think about the whole process
+	// 3. Can VMs later be added independently?
+	/*
+		vms, canceled, err := app.Prompt.CaptureListDecision(
+			app.Prompt,
+			"Provide a list of VMs this Subnet is running",
+			app.Prompt.CaptureEmpty,
+			"Provide a VM",
+			"VM",
+			"VMs are instances of blockchains a given Subnet is running.",
+			nil,
+		)
+		if err != nil {
+			return nil, err
+		}
+		if canceled {
+			ux.Logger.PrintToUser("Publishing aborted")
+			return nil, errors.New("canceled by user")
+		}
+
+		strVMs := make([]string, len(vms))
+		for i, v := range vms {
+			strVMs[i] = v.(string)
+		}
+	*/
+
+	subnet := &types.Subnet{
+		ID:          sc.Networks[models.Fuji.String()].SubnetID.String(),
+		Alias:       sc.Name,
+		Homepage:    homepage.(string),
+		Description: desc.(string),
+		Maintainers: strMaintrs,
+		// VMs:         strVMs,
+		VMs: nil,
+	}
+
+	return subnet, nil
+}
+
+func getVMInfo(sc *models.Sidecar) (*types.VM, error) {
+	vmID, err := app.Prompt.CaptureEmpty("What is the ID of this VM?", nil)
+	if err != nil {
+		return nil, err
+	}
+
+	desc, err := app.Prompt.CaptureEmpty("Provide a description for this VM please", nil)
+	if err != nil {
+		return nil, err
+	}
+
+	maintrs, canceled, err := app.Prompt.CaptureListDecision(
+		app.Prompt,
+		"Who are the maintainers of the VM?",
+		app.Prompt.CaptureEmail,
+		"Provide a maintainer",
+		"Maintainer",
+		"",
+		nil,
+	)
+	if err != nil {
+		return nil, err
+	}
+	if canceled {
+		ux.Logger.PrintToUser("Publishing aborted")
+		return nil, errors.New("canceled by user")
+	}
+
+	strMaintrs := make([]string, len(maintrs))
+	for i, m := range maintrs {
+		strMaintrs[i] = m.(string)
+	}
+
+	scr, err := app.Prompt.CaptureEmpty("What scripts needs to run to install this VM?", nil)
+	if err != nil {
+		return nil, err
+	}
+
+	bin, err := app.Prompt.CaptureEmpty("What is the binary path?", nil)
+	if err != nil {
+		return nil, err
+	}
+
+	url, err := app.Prompt.CaptureEmpty("Tell us the URL to download the binary please", nil)
```

Comment:
Might want to add needs to be a fixed version, not latest. And we need to download source, not a bin.

Reply:
Same